test/mem/remote.js

```diff
@@ -8,23 +8,20 @@ var log = require('../lib/log')
 var DB = require('../../lib/db/mem')(log, dbServer.errors)
 var P = require('bluebird')
 
-var server
-var db
-
-DB.connect(config)
+backendTests.remote(config, () => {
+  return DB.connect(config)
   .then(function (newDb) {
-    db = newDb
-    server = dbServer.createServer(db)
-    var d = P.defer()
+    const db = newDb
+    const server = dbServer.createServer(db)
+    const d = P.defer()
+    const close = server.close
+    server.close = () => {
+      close.call(server)
+      return db.close()
+    }
     server.listen(config.port, config.hostname, function() {
       d.resolve(server)
     })
     return d.promise
   })
-  .then(function(server) {
-    return backendTests.remote(config, server)
-  })
-  .then(function() {
-    server.close()
-    db.close()
-  })
+})
```

Comment:
I only just noticed that this module and `test/backend/remote.js` are identical except for the path to the db module in `require` on line 8. Should we take this opportunity to extract the duplicate code somewhere common?